src/types/context.rs

```diff
@@ -267,6 +277,16 @@ struct LockedContext<'ctx> {
 }
 
 impl<'ctx> LockedContext<'ctx> {
+    fn alloc_bound(&mut self, ctx_ptr: *const Mutex<Vec<Bound>>, bound: Bound) -> BoundRef {
```

Comment:
f2c7522df9ff50b5a262b6348ae3847326d36e13: How do we ensure that `ctx_ptr` points to the same memory as `self`?

This is a private method, so we can ensure that is being called correctly, but maybe it would be worth to add `*const Mutex<Vec<Bound>>` as a member to `LockedContext`. This would give `LockedContext` a consistent view of its context and we wouldn't have to pass `ctx_ptr` in `alloc_bound`.

Reply:
Sure. I think I had this in an earlier iteration.